main/EDInitMod.F90

```diff
@@ -384,51 +429,107 @@ subroutine init_patches( nsites, sites, bc_in)
+             newparea = area       
+           end if 
+
+           if(newparea.gt.0._r8)then ! Stop patches being initilialized when PFT not present in nocomop mode 
+              allocate(newp)
+
+              call create_patch(sites(s), newp, age, newparea, primaryforest, nocomp_pft)
+             
+              if(is_first_patch.eq.1)then !is this the first patch?
+                ! set poointers for first patch (or only patch, if nocomp is false)
+                newp%patchno = 1
+                newp%younger => null()
+                newp%older   => null()
+                sites(s)%youngest_patch => newp
+                sites(s)%oldest_patch   => newp
+                is_first_patch = 0
+              else ! the new patch is the 'oldest' one, arbitrarily. 
+                ! Set pointers for N>1 patches. Note this only happens when nocomp mode s on. 
+                ! The new patch is the 'youngest' one, arbitrarily.
+                newp%patchno = nocomp_pft
+                newp%older => recall_older_patch
+                newp%younger   => null()
+                recall_older_patch%younger => newp
+                sites(s)%youngest_patch   => newp
+              end if
+              recall_older_patch => newp ! remember this patch for the next one to point at. 
+
+              ! Initialize the litter pools to zero, these
+              ! pools will be populated by looping over the existing patches
+              ! and transfering in mass
+              do el=1,num_elements
+                 call newp%litter(el)%InitConditions(init_leaf_fines=0._r8, &
                    init_root_fines=0._r8, &
                    init_ag_cwd=0._r8, &
                    init_bg_cwd=0._r8, &
                    init_seed=0._r8,   &
                    init_seed_germ=0._r8)
-           end do
-           
-           sitep => sites(s)
-           call init_cohorts(sitep, newp, bc_in(s))
-           
-           ! For carbon balance checks, we need to initialize the 
-           ! total carbon stock
-           do el=1,num_elements
-              call SiteMassStock(sites(s),el,sites(s)%mass_balance(el)%old_stock, &
+             end do
+
+              sitep => sites(s)
+              call init_cohorts(sitep, newp, bc_in(s))
+            end if 
+         end do !no new patches
+         
+        !check if the total area adds to the same as site area
+         tota = 0.0_r8
+         newp => sites(s)%oldest_patch
+         do while (associated(newp))
+           tota=tota+newp%area
+             if ( debug ) write(fates_log(),*)  'test links',s,newp%nocomp_pft_label,tota
+           newp=>newp%younger
+         end do
+         if(abs(tota-area).gt.nearzero)then
+             write(*,*) 'error in assigning areas in init patch',s,tota-area
+         endif 
+
+          ! For carbon balance checks, we need to initialize the 
+          ! total carbon stock
+         write(*,*) 'calling sitemassstock',s
```

Comment:
recommend using:  if ( debug ) write(fates_log(),*) 'calling sitemassstock',s

Reply:
yeah i should've got rid of that one...